Alvaro Gaming/PythonGaming_p3.py

```diff
@@ -105,28 +140,49 @@ def move(self):
             sys.exit()
 
     DISPLAYSURF.blit(background, (0, 0))
-    scores = font_small.render(str(SCORE), True, BLACK)
-    DISPLAYSURF.blit(scores, (10, 10))
+    pygame.draw.line(background, WHITE, [380, 0], [380, 600], 5)
+    score_1 = font_small.render(str(SCORE_1), True, RED)
+    score_2 = font_small.render(str(SCORE_2), True, RED)
+    DISPLAYSURF.blit(score_1, (10, 10))
+    DISPLAYSURF.blit(score_2, (780, 10))
 
     # Moves and Re-draws all Sprites
     for entity in all_sprites:
         DISPLAYSURF.blit(entity.image, entity.rect)
         entity.move()
 
+    #Check if the ball is bouncing against any of the 4 walls:
+    if E1.rect.x>=750:
+        E1.velocity[0] = -E1.velocity[0]
+    if E1.rect.x<=0:
+        E1.velocity[0] = -E1.velocity[0]
+    if E1.rect.y>510:
+        E1.velocity[1] = -E1.velocity[1]
+    if E1.rect.y<0:
+        E1.velocity[1] = -E1.velocity[1]
```

Comment:
The players are handling their bounds within their Move function. It would be good to keep things standardized and have the ball do the same.

Reply:
They have been added to the move function so its constantly checking while in game loop by calling the Ball Class from the Sprite Group